linalg/densemat.cpp

```diff
@@ -3383,20 +3389,18 @@ void MultAAt(const DenseMatrix &a, DenseMatrix &aat)
 {
    const int height = a.Height();
    const int width = a.Width();
-   const DeviceVector d_a(a.GetData());
-   DeviceVector d_aat(aat.GetData());
-   MFEM_FORALL(i, height,
+   for (int i = 0; i < height; i++)
    {
-      for (int j=0; j<=i; j++)
+      for (int j = 0; j <= i; j++)
       {
-         double temp = 0.0;
-         for (int k=0; k<width; k++)
+         double temp = 0.;
+         for (int k = 0; k < width; k++)
          {
-            temp += d_a[i+k*height] * d_a[j+k*height];
+            temp += a(i,k) * a(j,k);
```

Comment:
Pretty cool that we can basically go back to the original code :+1:

Reply:
It's because this code doesn't run on GPU

Reply:
Yes, launching kernels for small `DenseMatrix` is not really a good idea, so I reverted the old code.